frontend/pages/packs/EditPackPage/EditPackPage.tsx

```diff
@@ -0,0 +1,371 @@
+    false
+  );
+
+  const [storedPackQueries, setStoredPackQueries] = useState<
+    IScheduledQuery[] | never[]
+  >([]);
+  const [isStoredPackQueriesLoading, setIsStoredPackQueriesLoading] = useState(
+    true
+  );
+  const [
+    isStoredPackQueriesLoadingError,
+    setIsStoredPackQueriesLoadingError,
+  ] = useState(false);
+
+  const getPack = useCallback(async () => {
+    setIsStoredPackLoading(true);
+    try {
+      const response = await packsAPI.load(packId);
+      setStoredPack(response.pack);
+    } catch (error) {
+      console.log(error);
+      setIsStoredPackLoadingError(true);
+    } finally {
+      setIsStoredPackLoading(false);
+    }
+  }, [dispatch]);
+
+  const getPackQueries = useCallback(async () => {
+    setIsStoredPackQueriesLoading(true);
+    try {
+      const response = await scheduledqueriesAPI.loadAll(packId);
+      setStoredPackQueries(response.scheduled);
+    } catch (error) {
+      console.log(error);
+      setIsStoredPackQueriesLoadingError(true);
+    } finally {
+      setIsStoredPackQueriesLoading(false);
+    }
+  }, [dispatch]);
+
+  useEffect(() => {
+    getPack();
+    getPackQueries();
+  }, [getPackQueries, getPack]);
+
+  const {
+    isLoading: isFleetQueriesLoading,
+    data: fleetQueries,
+    error: fleetQueriesError,
+  } = useQuery<IStoredFleetQueriesResponse, Error, IQuery[]>(
+    ["fleet queries"], // use single string or array of strings can be named anything
+    () => queriesAPI.loadAll(),
+    {
+      select: (data: IStoredFleetQueriesResponse) => data.queries,
+    }
+  );
+
+  const {
+    isLoading: isLabelsLoading,
+    data: labels,
+    error: packLabelsError,
+  } = useQuery<IStoredLabelsResponse, Error, ILabel[]>(
+    ["pack labels"], // use single string or array of strings can be named anything
+    () => labelsAPI.loadAll(),
+    {
+      select: (data: IStoredLabelsResponse) => data.labels,
+    }
+  );
+
+  const packLabels = storedPack
+    ? filter(labels, (label) => {
+        return includes(storedPack.label_ids, label.id);
+      })
+    : [];
+
+  const {
+    isLoading: isHostsLoading,
+    data: hosts,
+    error: hostsError,
+  } = useQuery<IStoredHostsResponse, Error, IHost[]>(
+    ["all hosts"], // use single string or array of strings can be named anything
+    () => hostsAPI.loadAll(undefined),
+    {
+      select: (data: IStoredHostsResponse) => data.hosts,
+    }
+  );
+  const packHosts = storedPack
+    ? filter(hosts, (host) => {
+        return includes(storedPack.host_ids, host.id);
+      })
+    : [];
+
+  const {
+    isLoading: isTeamsLoading,
+    data: teams,
+    error: teamsError,
+  } = useQuery<IStoredTeamsResponse, Error, ITeam[]>(
+    ["pack labels"],
+    () => teamsAPI.loadAll(),
+    {
+      select: (data: IStoredTeamsResponse) => data.teams,
+    }
+  );
+
+  const packTeams = storedPack
+    ? filter(teams, (team) => {
+        return includes(storedPack.team_ids, team.id);
+      })
+    : [];
```

Comment:
`packTeams` would be better as a local state, then you can use this logic to set it using the `onSuccess` event inside `useQuery`. This way, `teams` is always available and we increase efficiency. See the [docs for `useQuery`](https://react-query.tanstack.com/reference/useQuery).